gossip/c_block_callbacks_test.go

```diff
@@ -16,47 +17,51 @@ func TestConsensusCallback(t *testing.T) {
 	logger.SetTestMode(t)
 	require := require.New(t)
 
-	const blockCount = 100
+	const rounds = 30
 
-	env := newTestEnv()
+	const validatorsNum = 3
+
+	env := newTestEnv(2, validatorsNum)
 	defer env.Close()
 
-	accounts := len(env.validators)
+	accounts := validatorsNum
 
 	// save start balances
 	balances := make([]*big.Int, accounts)
 	for i := range balances {
-		balances[i] = env.State().GetBalance(env.Address(i + 1))
+		balances[i] = env.State().GetBalance(env.Address(idx.ValidatorID(i + 1)))
 	}
 
-	for n := uint64(0); n < blockCount; n++ {
+	for n := uint64(0); n < rounds; n++ {
 		// transfers
 		txs := make([]*types.Transaction, accounts)
 		for i := range txs {
 			from := (i)%accounts + 1
-			to := (i+1)%accounts + 1
-			txs[i] = env.Transfer(from, to, utils.ToFtm(100))
+			to := 0 + 1
```

Comment:
what you mean by writing  `0+1` instead of just `1`?

Reply:
`from` and `to` store validator IDs (normally typed as idx.ValidatorID), `i` and `accounts` store indexes of validators (normally typed as idx.Validator). Validator ID can never be 0. Normally (it depends on the SFC contract implementation) validator IDs start from 1 and increment after each new validator. In a general case, you cannot take validator index, add 1 and expect it to be his validator ID, because:
- Validators are sorted by stake first, so even a younger validator may have a smaller index than an older validator
- Some validators get deactivated, so validator IDs are sparse in a general case. E.g. we can have validator IDs `{1, 4, 9}`

In the test, we can just add 1 because all validators have the same stake and never get deactivated. Normally, one would need to call `from := env.store.GetValidators().GetID(idx.Validator((i)%accounts + 1))`. We write `0+1` instead of `1` to highlight that it's ID of first validator

In this test, the typing is not very strict, as both `i` and `accounts` are typed as int. It may make sense to update the typing to make it more obvious what variables store